### PR TITLE
add ambient occlusion to headlight

### PR DESCRIPTION
**Pull request by kevinyamauchi**
this adds an ambient occlusion tool to the mesh headlight

### CODE REVIEW

File: src/napari_threedee/visualization/lighting_control.py [hunk 1]
```diff
@@ -31,6 +34,58 @@ def _on_enable(self):
...
+
+            # set the data
+            # meshdata = visual.node._meshdata
+            # meshdata.set_vertex_values(attenuated_values)
+            # visual.node.set_data(meshdata=meshdata)
+
+    def _calc(self, vertices, faces, vertex_values):
+        vertex_normals = igl.per_vertex_normals(vertices, faces)
+        ao = igl.ambient_occlusion(vertices, faces, vertices, vertex_normals, 20)
+        attenuation_factor = 1 - ao
```
Comment (alisterburt) [inline src/napari_threedee/visualization/lighting_control.py:76]: formatting here is funky

**Review by alisterburt — APPROVED** [review]
Looking great @kevinyamauchi ! tiny qs but happy for you to merge when you're ready here

File: setup.cfg [hunk 2]
```diff
@@ -33,6 +33,7 @@ setup_requires = setuptools_scm
 install_requires =
 	einops
 	imageio>=2.5.0,!=2.11.0,!=2.22.1
+	libigl
```
[the remark this answers is not in the document]
Comment (kevinyamauchi) [reply setup.cfg:36]: Yup! pip works and conda-forge works for some platforms.

File: src/napari_threedee/visualization/lighting_control.py (hunk 1 above)
Comment (kevinyamauchi) [reply src/napari_threedee/visualization/lighting_control.py:76]: I'll clean this up. I need to do another pass to add some events to sync when the data gets updated.

**Pull request merged**